### PR TITLE
Fix the problem of the deployAddons.sh 

### PR DESCRIPTION
**Pull request by bjlee72**
This patch fixes the problem that the deployAddons.sh script does not install UI in Ubuntu environment.

### CODE REVIEW

**Comment by a-robinson** [comment]
Thanks @bjlee72!

@resouer @WIZARD-CXY does this look alright?

**Comment by a-robinson** [comment]
@k8s-bot ok to test

**Comment by WIZARD-CXY** [comment]
@bjlee72 Did you test it? I wonder why use `${KUBECTL} create -f namespace.yaml` twice?

File: cluster/ubuntu/deployAddons.sh [hunk 1]
```diff
@@ -16,19 +16,37 @@
...
+
+function deploy_dns {
+  echo "Deploying DNS on Kubernetes"
   sed -e "s/{{ pillar\['dns_replicas'\] }}/${DNS_REPLICAS}/g;s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g;" "${KUBE_ROOT}/cluster/addons/dns/skydns-rc.yaml.in" > skydns-rc.yaml
   sed -e "s/{{ pillar\['dns_server'\] }}/${DNS_SERVER_IP}/g" "${KUBE_ROOT}/cluster/addons/dns/skydns-svc.yaml.in" > skydns-svc.yaml
-  
+      
   # use kubectl to create kube-system namespace
-  "${KUBE_ROOT}/cluster/kubectl.sh" create -f namespace.yaml
+  ${KUBECTL} create -f namespace.yaml
```
Comment (WIZARD-CXY) [inline cluster/ubuntu/deployAddons.sh]: first in here.

File: cluster/ubuntu/deployAddons.sh [hunk 2]
```diff
@@ -16,19 +16,37 @@
...
   # use kubectl to create skydns rc and service
-  "${KUBE_ROOT}/cluster/kubectl.sh" --namespace=kube-system create -f skydns-rc.yaml
-  "${KUBE_ROOT}/cluster/kubectl.sh" --namespace=kube-system create -f skydns-svc.yaml
+  ${KUBECTL} --namespace=kube-system create -f skydns-rc.yaml 
+  ${KUBECTL} --namespace=kube-system create -f skydns-svc.yaml
+}
+
+function deploy_ui {
+  echo "Deploying Kubernetes UI"
+  ${KUBECTL} create -f namespace.yaml
```
Comment (WIZARD-CXY) [inline cluster/ubuntu/deployAddons.sh]: then in here

Comment (bjlee72) [inline cluster/ubuntu/deployAddons.sh]: At the bottom of the script file, there are two different if blocks. That means only one of the two functions can be executed. So, to make both `if` blocks configure the namespace safely, I redundantly put the create namespace command twice and removed the `set -e` flag: which means, if there is a namespace already created, though it shows error msg that the namespace is already there, but the script finishes OK.

Comment (WIZARD-CXY) [inline cluster/ubuntu/deployAddons.sh]: What if you create namespace in a new function like 'Init' or just create it before call `deploy_dns`or `deploy_ui`.  I'd like to keep the 'set -e' .

Comment (bjlee72) [inline cluster/ubuntu/deployAddons.sh]: In that case, to run the `./deployAddons.sh`, you should first `kube-down.sh`, and `kube-up.sh`, and finally `deployAddons.sh`. If it's the correct and expected behavior, I will change the script.

**Comment by bjlee72** [comment]
@WIZARD-CXY. I changed the code as commented with the following assumptions:

To run the ./deployAddons.sh, you should first run `kube-down.sh`, and `kube-up.sh` next, and finally `deployAddons.sh` ALWAYS. Thus, there is no way to run `deployAddons.sh` multiple times without shutting down the whole cluster.

**Comment by WIZARD-CXY** [comment]
OMG, why should we tear down the whole cluster to deploy addons again? Can we check if the `kube-system` namespace exists before creating it ? If it exists, we are not gonna create it?

**Comment by bjlee72** [comment]
Hmmm. if we do `set -e`, on `kubectl -f namespace.yaml` fails, the `deployAddons.sh` exists immediately. I thought that you know this issue. That's why I first removed `set -e` to prevent the whole script from exiting.

**Comment by bjlee72** [comment]
Anyway, as you said that my assumption is wrong, I will find another way :-)

**Comment by bjlee72** [comment]
Besides, if we leave `set -e`, then we cannot implement if-then-else logic. All the test that fails automatically makes the whole script exit. :-(

**Comment by bjlee72** [comment]
@WIZARD-CXY , I ask you to review the newly committed code one more time. As I already told you, it's impossible to implement the if-then-else logic that you requested in those previous comments with `set -e` (The reason I already explained). Thus, I implemented the if-then-else blocks without it. Though, the newly committed script basically does the same thing with the previous one, except for that it does not try re-install the addons if they are already installed.

**Comment by WIZARD-CXY** [comment]
Much better work, Let me test it and see if we can keep `set -e`

**Comment by WIZARD-CXY** [comment]
I understand your problem.It just like this one.
https://stackoverflow.com/questions/6550484/avoid-grep-returning-error-when-input-doesnt-match
I recommend keep using `set -e` to ensure it stops when other things fail and 
use `KUBEDNS=`eval "${KUBECTL} get services --namespace=kube-system | grep kube-dns | cat"`

**Comment by bjlee72** [comment]
Thanks @WIZARD-CXY . Following your suggestion, a new commit is issued.

**Comment by WIZARD-CXY** [comment]
Good job, I wonder will you update the doc by the way
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/ubuntu.md

**Comment by bjlee72** [comment]
@WIZARD-CXY , Thanks. I will update the doc, and will include that into the same pull request.

**Comment by WIZARD-CXY** [comment]
👍🏻👍🏻

**Comment by bjlee72** [comment]
`ubuntu.md` is updated.

**Comment by WIZARD-CXY** [comment]
Well done. LGTM

**Comment by a-robinson** [comment]
Thanks for reviewing, @WIZARD-CXY!

**Comment by a-robinson** [comment]
@k8s-bot test this please

**Comment by bjlee72** [comment]
Shippable build failed. Any hint to solve this?

**Comment by a-robinson** [comment]
You'll have to rebase your branch

**Comment by bjlee72** [comment]
All checks have passed. Thanks guys. I really learned a lot.

**Comment by piosz** [comment]
@bjlee72 Thanks for the change.

**Pull request merged**